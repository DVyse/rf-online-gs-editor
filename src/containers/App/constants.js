/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */
import { Map, List, fromJS } from 'immutable';

export const CHANGE_CURRENT_USER = 'global/CHANGE_CURRENT_USER';
export const CHANGE_CURRENT_USER_TOKEN = 'global/CHANGE_CURRENT_USER_TOKEN';
export const LOGOUT_CURRENT_USER = 'global/LOGOUT_CURRENT_USER';

// Projects Imports
export const PROJECTS_IMPORTS_CHANGE_PROP_VALUE =
  'global/PROJECTS_IMPORTS_CHANGE_PROP_VALUE';
export const PROJECTS_IMPORTS_START_FILE_IMPORT =
  'global/PROJECTS_IMPORTS_START_FILE_IMPORT';
export const PROJECTS_IMPORTS_CANCEL_FILE_IMPORT =
  'global/PROJECTS_IMPORTS_CANCEL_FILE_IMPORT';

// Projects Exports
export const PROJECTS_EXPORTS_CHANGE_PROP_VALUE =
  'global/PROJECTS_EXPORTS_CHANGE_PROP_VALUE';
export const PROJECTS_EXPORTS_START_FILE_EXPORT =
  'global/PROJECTS_EXPORTS_START_FILE_EXPORT';
export const PROJECTS_EXPORTS_CANCEL_FILE_EXPORT =
  'global/PROJECTS_EXPORTS_CANCEL_FILE_EXPORT';

// Projects NextValues
export const PROJECTS_NEXT_VALUES_CHANGE_PROP_VALUE =
  'global/PROJECTS_NEXT_VALUES_CHANGE_PROP_VALUE';
export const PROJECTS_NEXT_VALUES_CHANGE_NEXT_VALUE =
  'global/PROJECTS_NEXT_VALUES_CHANGE_NEXT_VALUE';
export const PROJECTS_NEXT_VALUES_CHANGE_NEXT_VALUE_ONLY_IN_STATE =
  'global/PROJECTS_NEXT_VALUES_CHANGE_NEXT_VALUE_ONLY_IN_STATE';
export const PROJECTS_NEXT_VALUES_CHANGE_IS_SAVING =
  'global/PROJECTS_NEXT_VALUES_CHANGE_IS_SAVING';
export const PROJECTS_NEXT_VALUES_CHANGE_IS_SAVED =
  'global/PROJECTS_NEXT_VALUES_CHANGE_IS_SAVED';
export const PROJECTS_NEXT_VALUES_CHANGE_IS_REMOVING =
  'global/PROJECTS_NEXT_VALUES_CHANGE_IS_REMOVING';
export const PROJECTS_NEXT_VALUES_CHANGE_IS_RESTORING =
  'global/PROJECTS_NEXT_VALUES_CHANGE_IS_RESTORING';
export const PROJECTS_NEXT_VALUES_CHANGE_IS_ERROR =
  'global/PROJECTS_NEXT_VALUES_CHANGE_IS_ERROR';
export const PROJECTS_NEXT_VALUES_CHANGE_IS_COPYING =
  'global/PROJECTS_NEXT_VALUES_CHANGE_IS_COPYING';
export const PROJECTS_NEXT_VALUES_CHANGE_ERROR_MESSAGE =
  'global/PROJECTS_NEXT_VALUES_CHANGE_ERROR_MESSAGE';
export const PROJECTS_NEXT_VALUES_REMOVE_VIRTUAL =
  'global/PROJECTS_NEXT_VALUES_REMOVE_VIRTUAL';
export const PROJECTS_NEXT_VALUES_RESTORE_VIRTUAL =
  'global/PROJECTS_NEXT_VALUES_RESTORE_VIRTUAL';

// Projects EntriesFinder
export const PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_WHERE_TYPE =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_WHERE_TYPE';
export const PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_WHERE_SEARCH =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_WHERE_SEARCH';
export const PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_SORT_BY =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_SORT_BY';
export const PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_SORT_WAY =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_SORT_WAY';
export const PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_TAKE_SKIP =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_FILTER_TAKE_SKIP';
export const PROJECTS_ENTRIES_FINDER_RESET_RESULT =
  'global/PROJECTS_ENTRIES_FINDER_RESET_RESULT';
export const PROJECTS_ENTRIES_FINDER_CHANGE_RESULT_ITEMS =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_RESULT_ITEMS';
export const PROJECTS_ENTRIES_FINDER_CHANGE_RESULT_TOTAL =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_RESULT_TOTAL';
export const PROJECTS_ENTRIES_FINDER_CHANGE_IS_LOADING =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_IS_LOADING';
export const PROJECTS_ENTRIES_FINDER_CHANGE_IS_LOADED =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_IS_LOADED';
export const PROJECTS_ENTRIES_FINDER_CHANGE_IS_ERROR =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_IS_ERROR';
export const PROJECTS_ENTRIES_FINDER_CHANGE_ERROR_MESSAGE =
  'global/PROJECTS_ENTRIES_FINDER_CHANGE_ERROR_MESSAGE';

// Action SubTypes
export const ITEM = 'ITEM';
export const STORE = 'STORE';
export const BOXITEMOUT = 'BOXITEMOUT';

// Immutable BaseObjects
export const IMMUTABLE_MAP = Map({});
export const IMMUTABLE_LIST = List([]);

// Projects EntriesFinder Defaults
export const PROJECTS_ENTRIES_FINDER_STATE_DEFAULTS = {
  [ITEM]: fromJS({
    subType: ITEM,
    isLoading: false,
    isLoaded: false,
    isError: false,
    errorMessage: '',
    filter: {
      take: 25,
      skip: 0,
      sortBy: 'nIndex',
      sortWay: 1,
      where: { search: '', type: '' },
    },
    result: { total: 1, items: [] },
  }),
  [STORE]: fromJS({
    subType: STORE,
    isLoading: false,
    isLoaded: false,
    isError: false,
    errorMessage: '',
    filter: {
      take: 25,
      skip: 0,
      sortBy: 'nIndex',
      sortWay: 1,
      where: { search: '' },
    },
    result: { total: 1, items: [] },
  }),
};

// Action States Constants
export const SKIP = 'skip';
export const REPLACE = 'replace';
export const WAITING = 'waiting';
export const PROCESSING = 'processing';
export const FINISHED = 'finished';
export const ERROR = 'error';
export const CANCELLED = 'cancelled';
export const LOADING = 'loading';

// Announces
export const ANNOUNCE_PROJECT_COUNT_ITEMS =
  'ANNOUNCE(s)/ANNOUNCE_PROJECT_COUNT_ITEMS';
export const ANNOUNCE_PROJECT_COUNT_STORES =
  'ANNOUNCE(s)/ANNOUNCE_PROJECT_COUNT_STORES';
export const ANNOUNCE_PROJECT_COUNT_BOXITEMOUTS =
  'ANNOUNCE(s)/ANNOUNCE_PROJECT_COUNT_BOXITEMOUTS';

// ProjectItem Name Fields (Ordered by priority)
export const PROJECT_ITEM_NAME_FIELDS = [
  ['priorStrName'],
  ['client', 'strName'],
  ['clientND', 'strName'],
  ['server', 'strName'],
];

// LocalSettings
export const AUTO_REVERSE_CLIENT_CODES = 'AUTO_REVERSE_CLIENT_CODES';
export const DISABLE_RENDER_ITEMS_IS_SCROLLING =
  'DISABLE_RENDER_ITEMS_IS_SCROLLING';
export const DISABLE_RENDER_ITEMS_IS_NOT_VISIBLE =
  'DISABLE_RENDER_ITEMS_IS_NOT_VISIBLE';
export const AUTO_RECALC_STORAGE_PRICE_IF_MONEY_VALUE_CHANGED =
  'AUTO_RECALC_STORAGE_PRICE_IF_MONEY_VALUE_CHANGED';
export const DEFAULT_STORAGE_PRICE_PERCENT = 'DEFAULT_STORAGE_PRICE_PERCENT';
