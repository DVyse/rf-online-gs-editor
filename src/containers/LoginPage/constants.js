/*
 *
 * LoginPage constants
 *
 */

export const DEFAULT_ACTION = 'app/LoginPage/DEFAULT_ACTION';
export const CHANGE_FIELD_VALUE = 'app/LoginPage/CHANGE_FIELD_VALUE';
export const SUBMIT = 'app/LoginPage/SUBMIT';
