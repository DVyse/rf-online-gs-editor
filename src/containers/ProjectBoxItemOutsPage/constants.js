/*
 *
 * ProjectBoxItemOutsPage constants
 *
 */

export const DEFAULT_ACTION = 'app/ProjectBoxItemOutsPage/DEFAULT_ACTION';
export const CHANGE_FILTER_TAKE_SKIP =
  'app/ProjectBoxItemOutsPage/CHANGE_FILTER_TAKE_SKIP';
export const CHANGE_FILTER_SORT_BY =
  'app/ProjectBoxItemOutsPage/CHANGE_FILTER_SORT_BY';
export const CHANGE_FILTER_SORT_WAY =
  'app/ProjectBoxItemOutsPage/CHANGE_FILTER_SORT_WAY';
export const CHANGE_FILTER_WHERE_SEARCH =
  'app/ProjectBoxItemOutsPage/CHANGE_FILTER_WHERE_SEARCH';
export const RESET_RESULT = 'app/ProjectBoxItemOutsPage/RESET_RESULT';
export const CHANGE_RESULT_TOTAL = 'app/ProjectBoxItemOutsPage/CHANGE_RESULT_TOTAL';
export const CHANGE_RESULT_ITEMS = 'app/ProjectBoxItemOutsPage/CHANGE_RESULT_ITEMS';
export const CONCAT_RESULT_ITEMS_FROM_JS =
  'app/ProjectBoxItemOutsPage/CONCAT_RESULT_ITEMS_FROM_JS';
