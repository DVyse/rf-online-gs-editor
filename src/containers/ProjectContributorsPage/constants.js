/*
 *
 * ProjectContributorsPage constants
 *
 */

export const DEFAULT_ACTION = 'app/ProjectContributorsPage/DEFAULT_ACTION';
export const CHANGE_ID = 'app/ProjectContributorsPage/CHANGE_ID';
export const CHANGE_FIELD_VALUE = 'app/ProjectContributorsPage/CHANGE_FIELD_VALUE';
export const CHANGE_PROJECT = 'app/ProjectContributorsPage/CHANGE_PROJECT';
