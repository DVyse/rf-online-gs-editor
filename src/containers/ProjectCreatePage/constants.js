/*
 *
 * ProjectCreatePage constants
 *
 */

export const DEFAULT_ACTION = 'app/ProjectCreatePage/DEFAULT_ACTION';
export const CHANGE_FIELD_VALUE = 'app/ProjectCreatePage/CHANGE_FIELD_VALUE';
export const SUBMIT = 'app/ProjectCreatePage/SUBMIT';
