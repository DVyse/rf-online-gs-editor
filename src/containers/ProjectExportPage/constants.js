/*
 *
 * ProjectExportPage constants
 *
 */

export const DEFAULT_ACTION = 'app/ProjectExportPage/DEFAULT_ACTION';
export const CHANGE_ID = 'app/ProjectExportPage/CHANGE_ID';
export const CHANGE_FIELD_VALUE = 'app/ProjectExportPage/CHANGE_FIELD_VALUE';
export const CHANGE_PROJECT = 'app/ProjectExportPage/CHANGE_PROJECT';
