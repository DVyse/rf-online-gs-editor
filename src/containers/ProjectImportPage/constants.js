/*
 *
 * ProjectImportPage constants
 *
 */

export const DEFAULT_ACTION = 'app/ProjectImportPage/DEFAULT_ACTION';
export const CHANGE_ID = 'app/ProjectImportPage/CHANGE_ID';
export const CHANGE_FIELD_VALUE = 'app/ProjectImportPage/CHANGE_FIELD_VALUE';
export const CHANGE_PROJECT = 'app/ProjectImportPage/CHANGE_PROJECT';
