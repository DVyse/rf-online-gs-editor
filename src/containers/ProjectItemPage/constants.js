/*
 *
 * ProjectItemPage constants
 *
 */

export const DEFAULT_ACTION = 'app/ProjectItemPage/DEFAULT_ACTION';
