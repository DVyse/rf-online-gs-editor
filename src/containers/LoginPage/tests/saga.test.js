/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { delay } from 'redux-saga';
import { all, call, take, put } from 'redux-saga/effects';

import {
  submit as submitAction,
  changeIsLoading,
  changeIsError,
  changeErrorMessage,
} from '../actions';

import defaultSaga, { watchSubmit, submit } from '../saga';
import { SUBMIT } from '../constants';
import {
  changeCurrentUserToken,
  changeCurrentUser,
} from '~/containers/App/actions';

jest.mock('redux-saga');

describe('defaultSaga Saga', () => {
  it('should have watchSubmit', () => {
    const generator = defaultSaga();
    const effect = generator.next().value;
    expect(effect).toEqual(all([call(watchSubmit)]));
  });
});

describe('watchSubmit Saga', () => {
  it('should call submit after SUBMIT action', () => {
    const generator = watchSubmit();
    const props = submitAction();
    expect(generator.next().value).toEqual(take(SUBMIT));
    expect(generator.next(props).value).toEqual(call(submit, props));
  });
});

describe('submit Saga', () => {
  it('should submit change state to loading', () => {
    const generator = submit();
    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().value).toEqual(put(changeIsLoading(true)));
    expect(generator.next().value).toEqual(put(changeIsError(false)));
    expect(generator.next().value).toEqual(put(changeErrorMessage('')));
  });
  it('should be delay 300 before mutation call', () => {
    const generator = submit();
    generator.next();
    generator.next();
    generator.next();
    generator.next();

    expect(generator.next().value).toEqual(delay(300));
  });
  it('should mutation must be ok', () => {
    const generator = submit();
    const loginPage = { ident: 'test', password: 'test' };

    generator.next();
    generator.next(loginPage);
    generator.next();
    generator.next();
    generator.next();

    // call apollo.mutate
    expect(generator.next().value).toMatchSnapshot();
    const result = {
      data: { userLogin: { token: '123', user: { login: '456' } } },
    };

    expect(generator.next(result).value).toEqual(put(changeIsLoading(false)));
    expect(generator.next().value).toEqual(
      put(changeCurrentUserToken(result.data.userLogin.token)),
    );
    expect(generator.next().value).toEqual(
      put(changeCurrentUser(result.data.userLogin.user)),
    );
  });
  it('should mutation must be fail', () => {
    const generator = submit();

    generator.next();
    generator.next();
    generator.next();
    generator.next();
    generator.next();

    // call apollo.mutate
    expect(generator.next().value).toEqual(put(changeIsError(true)));
    expect(generator.next().value).toEqual(
      put(changeErrorMessage("Cannot read property 'ident' of undefined")),
    );
    expect(generator.next().value).toEqual(put(changeIsLoading(false)));
  });
});
