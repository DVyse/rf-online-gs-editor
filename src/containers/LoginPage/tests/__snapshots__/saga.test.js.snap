// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submit Saga should mutation must be ok 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      Object {
        "mutation": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "UserLogin",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "ident",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "ident",
                          },
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "password",
                        },
                        "value": Object {
                          "kind": "Variable",
                          "name": Object {
                            "kind": "Name",
                            "value": "password",
                          },
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "userLogin",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "token",
                          },
                          "selectionSet": undefined,
                        },
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "user",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "login",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "role",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "name",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "title",
                                      },
                                      "selectionSet": undefined,
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "ident",
                    },
                  },
                },
                Object {
                  "defaultValue": undefined,
                  "kind": "VariableDefinition",
                  "type": Object {
                    "kind": "NonNullType",
                    "type": Object {
                      "kind": "NamedType",
                      "name": Object {
                        "kind": "Name",
                        "value": "String",
                      },
                    },
                  },
                  "variable": Object {
                    "kind": "Variable",
                    "name": Object {
                      "kind": "Name",
                      "value": "password",
                    },
                  },
                },
              ],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 264,
            "start": 0,
          },
        },
        "variables": Object {
          "ident": "test",
          "password": "test",
        },
      },
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`submit Saga should submit change state to loading 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;
